tests/plugins/test_deepspeed_plugin.py

```diff
@@ -0,0 +1,345 @@
+            assert isinstance(trainer.accelerator_backend.precision_plugin, DeepSpeedPrecisionPlugin)
+            assert trainer.accelerator_backend.precision_plugin.precision == 16
+            raise SystemExit()
+
+    model = BoringModel()
+    trainer = Trainer(fast_dev_run=True, plugins='deepspeed', callbacks=[CB()], amp_backend='native', precision=16)
+
+    with pytest.raises(SystemExit):
+        trainer.fit(model)
+
+
+@pytest.mark.skipif(not _DEEPSPEED_AVAILABLE, reason="DeepSpeed not available.")
+@pytest.mark.skipif(not _APEX_AVAILABLE, reason="Requires Apex")
+def test_deepspeed_apex_choice(tmpdir):
+    """
+        Test to ensure precision plugin is also correctly chosen. DeepSpeed handles precision via
+        Custom DeepSpeedPrecisionPlugin
+    """
+
+    class CB(Callback):
+
+        def on_fit_start(self, trainer, pl_module):
+            assert isinstance(trainer.accelerator_backend.training_type_plugin, DeepSpeedPlugin)
+            assert isinstance(trainer.accelerator_backend.precision_plugin, DeepSpeedPrecisionPlugin)
+            assert trainer.accelerator_backend.precision_plugin.precision == 16
+            raise SystemExit()
+
+    model = BoringModel()
+    trainer = Trainer(fast_dev_run=True, plugins='deepspeed', callbacks=[CB()], amp_backend='apex', precision=16)
+
+    with pytest.raises(SystemExit):
+        trainer.fit(model)
+
+
+@pytest.mark.skipif(not _DEEPSPEED_AVAILABLE, reason="DeepSpeed not available.")
+def test_deepspeed_with_invalid_config_path(tmpdir):
+    """
+        Test to ensure if we pass an invalid config path we throw an exception.
+    """
+
+    with pytest.raises(
+        MisconfigurationException, match="You passed in a path to a DeepSpeed config but the path does not exist"
+    ):
+        DeepSpeedPlugin(config='invalid_path.json')
+
+
+@pytest.mark.skipif(not _DEEPSPEED_AVAILABLE, reason="DeepSpeed not available.")
+def test_deepspeed_with_env_path(tmpdir, monkeypatch, deepspeed_config):
+    """
+        Test to ensure if we pass an env variable, we load the config from the path.
+    """
+    config_path = os.path.join(tmpdir, 'temp.json')
+    with open(config_path, 'w') as f:
+        f.write(json.dumps(deepspeed_config))
+    monkeypatch.setenv("DEEPSPEED_CONFIG_PATH", config_path)
+    plugin = DeepSpeedPlugin()
+    assert plugin.config == deepspeed_config
+
+
+@pytest.mark.skipif(not _DEEPSPEED_AVAILABLE, reason="DeepSpeed not available.")
+def test_deepspeed_defaults(tmpdir):
+    """
+    Ensure that defaults are correctly set as a config for DeepSpeed if no arguments are passed.
+    """
+    plugin = DeepSpeedPlugin()
+    assert plugin.config is not None
+    assert isinstance(plugin.config["zero_optimization"], dict)
+
+
+@pytest.mark.skipif(not torch.cuda.is_available(), reason="requires GPU machine")
+@pytest.mark.skipif(platform.system() == "Windows", reason="Distributed training is not supported on Windows")
```

Comment:
Why the windows check? No distributed here and DeepSpeed won't be available on windows

```suggestion
```